docs/DATA_MODEL.md

```diff
@@ -0,0 +1,299 @@
+
+One difference compared to Keys is the empty VariantValue, like `lang=`. This means "select all PropertyVariants which
+do NOT have the VariantName at all".
+Another difference is that `*` is allowed as a PropertyName, this would match and select all Properties. You can still
+add Variant Filters to the `*` filter.
+
+Let's look at some examples given following PropertyVariants of the `description` Property:
+
+* `description`
+* `description:lang=de`
+* `description:lang=de:format=markdown`
+* `description:lang=en`
+* `description:lang=en:format=markdown`
+
+With the filter `description:lang=de` you would get the two PropertyVariants `description:lang=de`
+and `description:lang=de:format=markdown`.
+
+With the filter `description` you would get all PropertyVariants as a result, because there is no filter except the
+PropertyName.
+
+With the empty VariantValue filter `description:lang=` you will only get `description` as a result, since this is the
+only PropertyVariant without the `lang` Variant.
+
+## Unknown Variants
+
+Boudicca will ignore all Variants it does not understand and treat the keys as if those Variants are not here. This
+behaviour is suggested for all Software handling Boudicca Keys.
+
+## Variants and the Search Service
+
+The Search Service will evaluate [Queries](QUERY.md) like the `contains` Query by searching all PropertyVariants
+matching the [KeyFilter](#keyfilters) given to the Query and then performing the Query against all the PropertyVariants.
+If any match, it will evaluate to true.
+The only Variant the Search Service does know about is the [Format Variant](#format-variant), because this will
+influence how searching. parsing and ordering works on those Variants.
+
+If you create invalid combinations of Queries and Variants (like an `contains` Query and a `format=date` Variant) Search
+Service will simply ignore those PropertyVariants as if they were not present.
+
+So lets say you have the following PropertyVariants:
+
+* `description`
+* `description:lang=de`
+* `description:lang=de:format=date`
+* `description:lang=en`
+* `description:lang=en:format=date`
+
+For the query `"description" contains "my search text"` Search Service will first select all PropertyVariants which
+match the given KeyFilter `description`, which are all of the above. Then it will discard all with an invalid format
+Variant, which leaves us with the PropertyVariants `description`, `description:lang=de` and `description:lang=en`. It
+will now evaluate the `contains` query against the value of those three PropertyVariants, and if any match the query
+will evaluate to true.
+
+## Variants and the HTML-Publisher
+
+For displaying properties with different Formats the HTML-Publisher follows
+the [Semantic Conventions](SEMANTIC_CONVENTIONS.md) and tries to find the Variant with the highest priority.
+
+If the user has selected a language (feature pending :P ) the Publisher will preferably select a PropertyVariant with
```

Comment:
[nit] hightlight this one if language select is visible in ui

Reply:
what do you mean?